commit_log/src/record.rs

```diff
@@ -0,0 +1,38 @@
+
+use super::segment::Segment;
+use super::position::Position;
+use std::io::Error;
+
+pub struct Record<'a> {
+    current_offset: usize,
+
+    segment: &'a mut Segment,
```

Comment:
I'm a but concerned of having the segment here, I understand why, but I'm trying to figure. 

I guess the "record" abstraction might be doing 2 things.

`Record` could be just the `&[u8]` wrapper, and we could have a `RecordReader` that then implements the `position`/`next`/...

WDYT?

Reply:
Yes, a `RecordReader` would make much more sense. But I am a little confused about what data should be wrapped under `Record`. A row pointer to the log or some other data format that we can design like [page table](https://www.wikiwand.com/en/Page_table#/Page_table_data)? Or something else?